src/commands/Register.ts

```diff
@@ -18,130 +18,215 @@ export type RegisterOptions = {
-    log.info('Executed Transactions', transactions);
+  if (isSameAddress(owner, from)) {
+    log.info('Relayer already authorized');
   }
-}
 
-export async function executeRegister(registerOptions?: RegisterOptions) {
+  const bal = await provider.getBalance(relayAddress);
+
+  if (bal.gt(options.funds)) {
+    log.info('Relayer already funded');
+  } else {
+    log.info('Funding relayer');
+
+    const fundTx = await options.signer.sendTransaction({
+      to: relayAddress,
+      value: options.funds,
+      gasLimit: 1e6,
+      gasPrice: options.gasPrice,
+    });
+
+    if (fundTx.hash == null) {
+      throw new Error(`Fund transaction reverted: ${JSON.stringify(fundTx)}`);
+    }
+    transactions.push(fundTx.hash);
+  }
+
+  await waitForRelay(httpClient, options.relayUrl);
+  log.info('Executed Transactions', transactions);
+};
+
+export async function executeRegister() {
+  // FIXME: add registerOptions?: RegisterOptions from either config or cli (which one?)
   const { app, contracts, blockchain } = getServerConfig();
-  let registerConfig: RegisterConfig | undefined = undefined;
-  if (config.has('register')) {
-    registerConfig = config.get('register');
+  const { account, stake, funds, mnemonic }: RegisterConfig = config.has(
+    'register'
+  )
+    ? config.get<RegisterConfig>('register')
+    : {};
+
+  console.log(
+    `🐞 ᨟ ---------------------------------------------------------------------------------------------------------------------------🐞 ᨟`
+  );
+  console.log(
+    `🐞 ᨟ :: file: register.ts:176 :: executeRegister :: { account, stake, funds, mnemonic }`,
+    { account, stake, funds, mnemonic }
+  );
+  console.log(
+    `🐞 ᨟ ---------------------------------------------------------------------------------------------------------------------------🐞 ᨟`
+  );
+
+  if (account && !mnemonic) {
+    log.error(`
+    You must configure mnemonic for given account address.
+    `);
   }
+
   log.setLevel(app.logLevel);
-  const register = new Register(
-    blockchain.rskNodeUrl,
-    registerConfig?.mnemonic
-  );
+  const rpcProvider = new JsonRpcProvider(blockchain.rskNodeUrl);
   const portIncluded: boolean = app.url.indexOf(':') > 0;
   const relayUrl =
     app.url + (!portIncluded && app.port > 0 ? ':' + app.port.toString() : '');
-  await register.execute(
-    registerOptions
-      ? registerOptions
-      : {
-          hub: contracts.relayHubAddress,
-          signer: await register.findWealthyAccount(),
-          stake: utils.parseEther(registerConfig?.stake ?? '0.01'),
-          funds: utils.parseEther(registerConfig?.funds ?? '0.02'),
-          relayUrl,
-          unstakeDelay: '1000',
-          gasPrice: '60000000',
-        }
+
+  const signer =
+    account && mnemonic
+      ? Wallet.fromMnemonic(mnemonic).connect(rpcProvider)
+      : await findWealthyAccount(rpcProvider);
+
+  const signerAddress = await signer.getAddress();
+  if (signerAddress !== account) {
```

Comment:
This validation would not allow to use the findWealthyAccount, unless we know which account is going be found.

Reply:
rookie mistake... thanks for spotting it

Reply:
done